src/models.h

```diff
@@ -696,21 +728,139 @@ namespace tobor {
 
 			using vector_type = std::vector<piece_move_type>;
 
+			using pieces_quantity_type = typename piece_move_type::pieces_quantity_type;
+
 		private:
 			vector_type move_vector;
 
 		public:
 
 			move_path() {}
 
+			move_path(std::size_t n) : move_vector(n, piece_move_type()) {}
+
 			vector_type& vector() { return move_vector; }
 
+			const vector_type& vector() const { return move_vector; }
+
 			inline move_path operator +(const move_path& another) {
 				move_path copy{ *this };
 				std::copy(another.move_vector.cbegin(), another.move_vector.cend(), std::back_inserter(copy.move_vector));
 				return copy;
 			}
 
+			/*
+			inline std::vector<bool> colored_footprint() const {
+
+				std::array<vector_type::const_iterator, pieces_quantity_type::COUNT_ALL_PIECES> color_next;
+
+				for (auto& iter : color_next) {
+					iter = move_vector.cbegin();
+				}
+
+				while (true) {
+					uint8_t i = 0;
+					bool IS_NORTH_OR_SOUTH = 0;
+					bool
+					while (color_next[i] != move_vector.cend() && color_next[i]->pid.value != i) {
+						++color_next[i];
+					}
+					if (color_next[i] != move_vector.cend()) {
+
+						++color_next[i];
+					}
+				}
+
+				... to much effort in coding for a footprint. will also probably take longer computation time.
+			}
+			*/
+
+			inline move_path color_sorted_footprint() const {
```

Comment:
unused

Check time consuption for state graph exploration versus equivalence class analysis.
Check optimization potential for equivalence class construction

Reply:
(#85 )